src/Symfony/Component/Serializer/Serializer.php

```diff
@@ -183,6 +193,14 @@ public function normalize($data, $format = null, array $context = [])
      */
     public function denormalize($data, $type, $format = null, array $context = [])
     {
+        if (isset(self::SCALAR_TYPES[$type])) {
+            if (false === @settype($data, $type)) {
+                throw new LogicException(sprintf('"%s" cannot be denormalized to %s', (string) $data, $type));
```

Comment:
the cast to string can fail - why no using gettype+get_class?

Reply:
Could you write an example of exception message? This will afford me to better understand what you mean.

Reply:
sure:
`throw new LogicException(sprintf('Value of type "%s" cannot be denormalized to "%s".', \gettype($data), $type)); `